src/object.d

```diff
@@ -1703,7 +1703,7 @@ class TypeInfo_Class : TypeInfo
     override size_t getHash(scope const void* p) @trusted const
     {
         auto o = *cast(Object*)p;
-        return o ? o.toHash() : 0;
+        return hashOf(o ? o.toHash() : 0);
```

Comment:
Why is this necessary instead of
```d
o ? o.toHash() : 0
```
?
The reason that the version in `core.internal.hash.hashOf` is different is that it (currently) doesn't distinguish between the chaining and non-chaining cases, but that isn't the case here.

Reply:
Instead I'd write just `return hashOf(o)` if the goal is to ensure that the result is the same as `core.internal.hashOf`.